### DISCUSSION TITLE
What is the right way to delete an IP or host?

### DISCUSSION THREAD

**Discussion by john2893** [netbox-community/netbox#14907] [Help Wanted!] [answered]
ip = IPAddress(
                address='192.168.3.10/32',
                dns_name='test.xyz.com'
        )
        my_obj1 = IPAddress.objects.get(dns_name='test.xyz.com')

        
       ================
       ip.delete()   -- Does not work. 
       my_obj1.delete() -- Does not work. 
       IPAddress.objects.filter(dns_name='test.xyz.com').delete()  -- Does not work. 

I know I'm missing something but just can't figure what it is. I want to do it right. 
I just learnt from saving an IP also needs full_clean() before saving. So, I wanted to do the deleting clean and right

**Comment by jeremystretch** [comment]
`ip = IPAddress(...)` creates the instance but does not save it to the database. You can call `ip.full_clean()` to validate the instance and `ip.save()` to save it, after which calling `ip.delete()` will delete it from the database.

**Reply by john2893** [reply]
Thank you!

I did try ip.delete() 
I get "ValueError: IPAddress object can't be deleted because its id attribute is set to None."

This IP and hostname does show in the netbox GUI

and in db:

```
netbox=> select id, address, status, role, assigned_object_id, dns_name, assigned_object_type_id from ipam_ipaddress where address='192.168.3.10/32';
  id       |   address       | status | role | assigned_object_id |       dns_name        | assigned_object_type_id 
-------+--------------+--------+------+--------------------+-----------------------+-------------------------
 16997 | 192.168.3.10 | active |      |                                  |  test.xyz.com          |                        
(1 row)

```

**Comment by candlerb** [comment]
If you create an brand new object but don't save it, then it's just a local object in RAM.

If you want to delete an object which is already in the database, then first read it in using `find()` or `get()`, and then delete it.

```
ip = IPAddress.objects.get(address='192.168.3.10/32')
ip.delete()
```

**Reply by john2893** [reply, marked as answer]
Thank you!

I had to restart netbox services and now it works.!
